executor/src/witgen/mod.rs

```diff
@@ -157,54 +108,13 @@ where
         .committed_polys_in_source_order()
         .into_iter()
         .map(|(p, _)| {
-            let column = secondary_columns
-                .remove(&p.absolute_name)
-                .unwrap_or_else(|| {
-                    let column = &mut columns[&PolyID {
-                        id: p.id,
-                        ptype: PolynomialType::Committed,
-                    }];
-                    std::mem::take(column)
-                });
+            let column = columns.remove(&p.absolute_name).unwrap();
             assert!(!column.is_empty());
             (p.absolute_name.as_str(), column)
         })
         .collect()
 }
 
-fn zip_relevant<'a, T>(
```

Comment:
Seems like I removed this, now loop detection looks at all columns, even those of other machines (which will always be empty, and hence equal).

I think think that's fine! Loop detection only runs every 100 rows.
In the future, we should also add the possibility for `ColumnMap` to only represent a subset of the columns (#605), which will have efficiency gains in other places as well.

Only doing loop detection on the "relevant" columns was introduced by me in #460. I don't remember any justification for this and there is none in the PR description.